backend/controller/observability/deployment.go

```diff
@@ -0,0 +1,82 @@
+package observability
+
+import (
+	"context"
+	"fmt"
+
+	"go.opentelemetry.io/otel"
+	"go.opentelemetry.io/otel/attribute"
+	"go.opentelemetry.io/otel/metric"
+
+	"github.com/TBD54566975/ftl/internal/observability"
+)
+
+const (
+	deploymentMeterName = "ftl.deployments.controller"
+)
+
+type DeploymentMetrics struct {
+	reconciliationFailures metric.Int64Counter
+	reconciliationsActive  metric.Int64UpDownCounter
+	replicasAdded          metric.Int64Counter
```

Comment:
Since we increase and lower this value, should it be an UpDown counter instead? I have no idea btw - in the middle of reading more about this myself

Reply:
You are right! I went with counter given `replicasAdded` is a delta measure. However the regular `counter` expects monotonicity - so I am required to use the `up-down-counter` in this case. Per this documentation it is valid to use it for a delta measure: https://opentelemetry.io/docs/specs/otel/metrics/supplementary-guidelines/.

Nice catch, fix applied.

Alternatively, it might be better to track additions and reductions separately (wdyt?)

Reply:
sweet, thanks for digging that up! :D

Reply:
Are we also exporting "replicas" somewhere as a gauge?